### PR TITLE
Use the chunk id instead of the chunk name in the cssAssets array

### PR DESCRIPTION
**Pull request by tjfred35**
fixes: https://github.com/single-spa/single-spa-css/issues/11

`cssAssetFileNames` is a function that is generated by `GetChunkFilenameRuntimeModule`, which returns a function whose parameter name is chunkId.

In some builds this doesn't matter, because the output of the `cssAssetFileNames` minifies to something like: 
```
i.cssAssetFileName=e=>"main.4b69191f341ccadb8069.css"
```

making the actual value stored in `cssAssets` irrelevant.

But when using React.lazy, GetChunkFilenameRuntimeModule produces a more complicated function:

```
i.cssAssetFileName=e=>(179===e?"main":e)+"."+{179:"4b69191f341ccadb8069"}[e]+".css"
```

where the value of the parameter (chunkId / e) matters. Because we write the chunk _name_ to `cssAssets`, we break.

This has something to do with the fact that there is more than one chunk evaluated by GetChunkFilenameRuntimeModule.

We could sort of solve this problem like this

```
compilation.addRuntimeModule(
  chunk,
  new webpack.runtime.GetChunkFilenameRuntimeModule(
    MODULE_TYPE,
    "single-spa-css",
    `${webpack.RuntimeGlobals.require}.cssAssetFileName`,
    (referencedChunk) => {
      if (!referencedChunk.contentHash[MODULE_TYPE]) {
        return false;
      }

      return referencedChunk.canBeInitial()
        ? this.options.filename
        : this.options.chunkFilename;
    },
    true
  )
);
```

This seems more in line with the usage of GetChunkFilenameRuntimeModule in mini-css-extract-plugin,

but it only solves it by making the parameter value of cssAssetFileName irrelevant again:

```
i.cssAssetFileName=e=>"main.4b69191f341ccadb8069.css"
```

We could leave cssAssets alone and create (and use) a new cssAssetIds array instead, but I wasn't sure what the philosophy was around that sort of thing, so i opted for this simpler change.

### CODE REVIEW

**Comment by tjfred35** [comment]
@joeldenning Can you take a look at this?

**Comment by joeldenning** [comment]
I will look at this soon, maybe tomorrow. Have been unavailable for a week or so and am just am catching up on github notifications.

**Comment by joeldenning** [comment]
Thanks for the explanation. I have authored a few webpack plugins but they aren't my strongest area so it took me a few times to read through the explanation and code to understand everything going on here.

We should definitely support multiple css chunks/files, including those created via `import()`.

Looking over the code, it seems changing cssAssets to store chunk ids rather than chunk names is the best approach and something I just overlooked/missed in my original implementation. Those changes look good.

But it seems that the current changes you've proposed would work only if the `chunkFilename` is equivalent to the `filename`? Don't we need to also make the change you suggested above that adds a `chunkFilename` option, to properly account for situations where the main css file and chunk css files have been configured for different file names?

Regarding that approach, you said the following:

> This seems more in line with the usage of GetChunkFilenameRuntimeModule in mini-css-extract-plugin, but it only solves it by making the parameter value of cssAssetFileName irrelevant again

My understanding of GetChunkFilenameRuntimeModule (which might be wrong or incomplete) is that it generates a `cssAssetFileName` function hard coded to one chunk if there is only one chunk, but that it generates a function that's dynamic to the chunk id if there are multiple chunks. Is that not what you were seeing in your testing?

In other words, I think we should add `return referencedChunk.canBeInitial() ? this.options.filename : this.options.chunkFilename;` to where we're calling `GetChunkFilenameRuntimeModule`. Let me know if there are issues with that approach - I think we'd publish it as a major version of single-spa-css as it would require users to pass in a new option `chunkFilename` that previously was not required.

**Comment by tjfred35** [comment]
I didn't have a test case for multiple CSS files, so I didn't choose to introduce support for chunk name for css files. We really only have multiple js files for our test case. I thought about seeing if I could get it working, since `[ this.chunk.id ]` felt wrong, but I couldn't imagine a case where we would want multiple css files, but still load them all at bootstrap / mount.

**Comment by joeldenning** [comment]
👍🏿  sounds good, I think one use case for multiple CSS files would be if they are content hashed and one includes vendor css and another application css - since the vendor one wouldn't change as frequently you could get the performance benefit of browser caching the vendor css file while still updating the application css.
